dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/dataset/notifications/DefaultDataSetNotificationService.java

```diff
@@ -116,34 +118,51 @@
     // Dependencies
     // -------------------------------------------------------------------------
 
-    @Autowired
-    private DataSetNotificationTemplateService dsntService;
+    private final DataSetNotificationTemplateService dsntService;
 
-    @Autowired
-    private MessageService internalMessageService;
+    private final MessageService internalMessageService;
 
-    @Autowired
-    private ProgramMessageService externalMessageService;
+    private final ProgramMessageService externalMessageService;
 
-    @Autowired
-    private NotificationMessageRenderer<CompleteDataSetRegistration> renderer;
+    private final NotificationMessageRenderer<CompleteDataSetRegistration> renderer;
 
-    @Autowired
-    private CompleteDataSetRegistrationService completeDataSetRegistrationService;
+    private final CompleteDataSetRegistrationService completeDataSetRegistrationService;
 
-    @Autowired
-    private PeriodService periodService;
+    private final PeriodService periodService;
 
-    @Autowired
-    private CategoryService categoryService;
+    private final CategoryService categoryService;
 
-    @Autowired
-    private I18nManager i18nManager;
+    private final I18nManager i18nManager;
+
+    private final OrganisationUnitService organisationUnitService;
 
     @Autowired
-    private OrganisationUnitService organisationUnitService;
+    public DefaultDataSetNotificationService( DataSetNotificationTemplateService dsntService, MessageService internalMessageService, ProgramMessageService externalMessageService,
+         NotificationMessageRenderer<CompleteDataSetRegistration> renderer, CompleteDataSetRegistrationService completeDataSetRegistrationService, PeriodService periodService,
+         CategoryService categoryService, I18nManager i18nManager, OrganisationUnitService organisationUnitService )
+    {
+        checkNotNull( dsntService );
```

Comment:
I dont think you need to check for null here. If they are null, what are you gonna do? We should assume it works in my opinion.

Reply:
I think earlier to catch Null pointer the better. Otherwise, the logic of the class might do some harm before this service executes any method.